resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -39,57 +40,107 @@ function Get-ValidTimeZone {
+    Whether to set the time zone automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time zone automatically.
+
+.PARAMETER SetTimeAutomatically
+    Whether to set the time automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time automatically.
 
 .PARAMETER AdjustForDaylightSaving
     Whether to adjust for daylight saving time. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Adjust for daylight saving time automatically.
 
 .EXAMPLE
-    PS C:\> Invoke-DscResource -Name Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
+    PS C:\> Invoke-DscResource -Name TimeZone -ModuleName Microsoft.Windows.Setting.Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
 
     This example sets the time zone to Pacific Standard Time.
 
 .EXAMPLE
-    PS C:\> Invoke-DscResource -Name Time -Method Get -Property {}
+    PS C:\> Invoke-DscResource -Name TimeZone -ModuleName Microsoft.Windows.Setting.Time -Method Get -Property {}
 
     This example gets the current time settings on the machine.
 #>
 [DscResource()]
 class TimeZone {
     [DscProperty(Key)]
-    [string] $TimeZone
+    [string] $Id
 
     [DscProperty()]
     [nullable[bool]] $SetTimeZoneAutomatically
 
+    [DscProperty()]
+    [nullable[bool]] $SetTimeAutomatically
+
     [DscProperty()]
     [nullable[bool]] $AdjustForDaylightSaving
 
-    static hidden [string] $SetTimeZoneAutomaticallyProperty = 'Type'
+    static hidden [string] $SetTimeZoneAutomaticallyProperty = 'Start'
+    static hidden [string] $SetTimeAutomaticallyProperty = 'Type'
     static hidden [string] $AdjustForDaylightSavingProperty = 'DynamicDaylightTimeDisabled'
+    static hidden [string] $NtpEnabled = 'NTP'
+    static hidden [string] $NtpDisabled = 'NoSync'
+    static hidden [bool] $SupportsDaylightSavingProperty = $false
 
     TimeZone() {
-        $this.TimeZone = (Get-TimeZone).Id
+        $timeZone = Get-TimeZone
+        $this.Id = $timeZone.Id
+
+        # We set the SupportsDaylightSavingProperty to the value that is supported by the current time zone
+        [TimeZone]::SupportsDaylightSavingProperty = $timeZone.SupportsDaylightSavingTime ? $true : $false
     }
 
     [TimeZone] Get() {
         $currentState = [TimeZone]::new()
-        $currentState.SetTimeZoneAutomatically = [TimeZone]::GetTimeZoneAutoUpdateStatus()
-        $currentState.AdjustForDaylightSaving = [TimeZone]::GetDayLightSavingStatus()
+        $currentState.SetTimeZoneAutomatically = [TimeZone]::GetTimeZoneAutomaticallyStatus()
+        $currentState.SetTimeAutomatically = [TimeZone]::GetTimeAutomaticallyStatus()
+
+        if ($currentState::SupportsDaylightSavingProperty) {
+            $currentState.AdjustForDaylightSaving = [TimeZone]::GetDayLightSavingStatus()
+        }
```

Comment:
This value is not dependent on whether the time zone supports daylight saving or not. This should just be a registry check.

Reply:
I didn't totally get you here. It is checking the registry with the `GetDayLightSavingStatus()` method?